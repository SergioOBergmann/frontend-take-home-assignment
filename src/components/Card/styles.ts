import styled from 'styled-components';

import { MOBILE_BREAKPOINT, PrimaryButton } from '../../styles/global';

export const CardDiv = styled.div`
  flex-direction: column;
  justify-content: center;
  margin: 40px auto auto auto;
  padding: 40px;
  width: 478px;
  height: 518px;

  background: #ffffff;
  border: 1px solid #e1e8ed;
  box-shadow: 0px 1px 4px rgba(150, 164, 176, 0.1);
  border-radius: 8px;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    width: calc(100% - 48px - 2px);
    min-width: 310px;
    height: 100%;
    margin: 24px auto auto auto;
    padding: 24px;
  }
`;

export const Icon = styled.img`
  width: 40px;
  height: 40px;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    width: 32px;
    height: 32px;
    margin-top: 4px;
  }
`;

export const Title = styled.span`
  margin-top: 4px;

  font-weight: 600;
  font-size: 32px;
  line-height: 40px;
  letter-spacing: -0.266667px;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    font-size: 24px;
    line-height: 32px;
    letter-spacing: -0.2px;
  }
`;

export const Display = styled.span`
  margin-top: 2px;

  font-size: 16px;
  line-height: 20px;

  color: #657786;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    margin-top: 0;
    font-size: 14px;
  }
`;

export const CardButton = styled(PrimaryButton as any)`
  margin: 31px 40px auto 40px;

  @media (max-width: ${MOBILE_BREAKPOINT}px) {
    margin: 36px -8px auto -8px;
  }
`;
