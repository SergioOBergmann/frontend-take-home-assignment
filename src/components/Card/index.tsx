import * as React from 'react';
import { CardDiv, Icon, Title, Display, CardButton } from './styles';

type CardProps = {
  iconImg: any;
  iconText: string;
  title: string;
  display?: string;
  buttonText?: string;
  children?: React.ReactNode;
};

const Card = ({
  iconImg,
  iconText,
  title,
  display = 'Saving Goal',
  buttonText = 'Confirm',
  children
}: CardProps): JSX.Element => {
  return (
    <>
      <CardDiv>
        <Icon src={iconImg} alt={iconText} />
        <Title>{title}</Title>
        <Display>{display}</Display>

        {children}

        <CardButton>{buttonText}</CardButton>
      </CardDiv>
    </>
  );
};

export default Card;
