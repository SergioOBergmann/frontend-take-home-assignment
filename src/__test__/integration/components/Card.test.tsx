import * as React from 'react';

import { render, RenderResult } from '@testing-library/react';

import Card from '../../../components/Card';

let documentBody: RenderResult;

describe('<Card />', () => {
  beforeEach(() => {
    documentBody = render(
      <Card iconImg="" iconText="House" title="Buy a house" />
    );
  });

  it('matches snapshot', () => {
    const { baseElement } = documentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
