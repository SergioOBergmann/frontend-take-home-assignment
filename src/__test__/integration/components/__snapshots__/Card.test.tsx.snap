// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> matches snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-gzVnrw fGSdvJ"
    >
      <img
        alt="House"
        class="sc-htoDjs jMUYuv"
        src=""
      />
      <span
        class="sc-dnqmqq fvEnnj"
      >
        Buy a house
      </span>
      <span
        class="sc-iwsKbI hNxSrM"
      >
        Saving Goal
      </span>
      <button
        class="sc-bZQynM sc-gZMcBi dNYqZl"
      >
        Confirm
      </button>
    </div>
  </div>
</body>
`;
